src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/PhoneAttribute.cs

```diff
@@ -31,7 +27,72 @@ public override bool IsValid(object value)
             }
 
             var valueAsString = value as string;
-            return valueAsString != null && _regex.Match(valueAsString).Length > 0;
+            if (valueAsString == null)
+            {
+                return false;
+            }
+
+            valueAsString = valueAsString.Replace("+", string.Empty).TrimEnd();
+            valueAsString = RemoveExtension(valueAsString);
+
+            if (!valueAsString.Any(Char.IsDigit))
+            {
+                return false;
+            }
+
+            return valueAsString.All(c =>
+                Char.IsDigit(c)
+                || Char.IsWhiteSpace(c)
+                || _additionalPhoneNumberCharacters.Contains(c));
+        }
+
+        private static string RemoveExtension(string potentialPhoneNumber)
+        {
+            var lastIndexOfExtension = potentialPhoneNumber
+                .LastIndexOf("ext.", StringComparison.OrdinalIgnoreCase);
+            if (lastIndexOfExtension >= 0)
+            {
+                var extension = potentialPhoneNumber.Substring(lastIndexOfExtension + 4);
```

Comment:
Instead of having these magic numbers, could we put the strings being used in consts and then use .Length off of those?